Lib/test/test_asyncio/test_base_events.py

```diff
@@ -1590,7 +1590,8 @@ def test_create_datagram_endpoint_no_addrinfo(self, m_socket):
             MyDatagramProto, local_addr=('localhost', 0))
         self.assertRaises(
             OSError, self.loop.run_until_complete, coro)
-
+            
+    @unittest.skipIf(sys.flags.optimize, "Assertions are disabled in optimized mode")
```

Comment:
It is not correct. The code should raise a TypeError. I think a simple `raise TypeError('2-tuple is expected')` should be enough, there are precedents in the same function. 

Other tests can be skipped, they test internal methods.

Reply:
fixed!